my-code/frontend/src/pages/Movie/index.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { useParams } from "react-router";
+import { IMovieDetails } from "./types";
+import api from "../../services/api";
+import BackButton from "../../components/BackButton";
+import {
+  Container,
+  MovieInfoContainer,
+  Poster,
+  PlotContainer,
+  CastGenreDirector,
+} from "./styles";
+import Rating from "../../components/Rating";
+import FavouriteButton from "../../components/FavouriteButton";
+import SearchError from "../../components/SearchError";
+import LoadingSearch from "../../components/LoadingSearch";
+interface IParams {
+  id: string;
+}
+
+enum RequestStatus {
+  Idle,
+  Loading,
+  Loaded,
+  Error,
+}
+
+const Movie: React.FC = () => {
+  const [status, setStatus] = React.useState(RequestStatus.Idle);
+  const [details, setDetails] = React.useState<IMovieDetails | undefined>();
+  const params: IParams = useParams();
+  React.useEffect(() => {
+    (async function () {
+      try {
+        setStatus(RequestStatus.Loading);
+        const { data }: { data: IMovieDetails } = await api.get(
+          `/movie/${params.id}`
+        );
+        if (data.Response === "False") {
+          setStatus(RequestStatus.Error);
+          return;
+        }
+
+        setDetails(data);
+        setStatus(RequestStatus.Loaded);
+      } catch {
+        setStatus(RequestStatus.Error);
+      }
+    })();
+  }, []);
```

Comment:
Missing dependencies

```suggestion
  }, [setStatus, setDetails, params]);
```